### ISSUE TITLE
'Expression' object has no attribute 'tablename'

### ISSUE THREAD

**Issue by john11721** [web2py/web2py#1735]
Has this problem been resolved:

https://groups.google.com/forum/#!topic/web2py/-6Xz48ECcEM

Still see this on
web2py™ | Version 2.15.3
-- | --
 rows = db(q).select(t.ALL, max, groupby=t.ALL, join=join) # this produces the error above :(
 rows = db(q).select(t.ALL, max, groupby=t.ALL, join=join).as_list() #this doesn't throw the error :)

**Comment by dburba** [comment]
This error affects me as well (with PostgreSQL)

Version 2.15.4-stable+timestamp.2017.09.02.04.02.22

```
db.define_table('measurement',
    Field('title', label=T('measurement__title'), requires=IS_NOT_EMPTY()),
    Field('systemic', 'boolean', writable=False, readable=False, default=False),
    Field('sys_name', writable=False, readable=False),
    format='%(title)s'
)

title_field = db.translation_field.value.coalesce(db.measurement.title)
TOTAL_ROWS = 'COUNT(*) OVER()'

records = db(db.measurement).select(
        db.measurement.ALL, title_field, TOTAL_ROWS,
        left=[
            db.translation_field.on(
                (db.translation_field.tablename == 'measurement') &
                (db.translation_field.fieldname == 'title') &
                (db.translation_field.rid == db.measurement.id) &
                (db.translation_field.language_id == auth.user.language_id)
            )
        ],
        orderby=title_field
    )

# translate
for i, r in enumerate(records):
    records[i].measurement.title = r[title_field]

result = SQLTABLE(records, headers='labels', truncate=None, columns=['measurement.title')
```
Error:

```
Traceback (most recent call last):
  File "/vagrant/grand3/web2py/gluon/restricted.py", line 219, in restricted
    exec(ccode, environment)
  File "/vagrant/grand3/web2py/applications/app/controllers/admin.py", line 695, in <module>
  File "/vagrant/grand3/web2py/gluon/globals.py", line 414, in <lambda>
    self._caller = lambda f: f()
  File "/vagrant/grand3/web2py/gluon/tools.py", line 3981, in f
    return action(*a, **b)
  File "/vagrant/grand3/web2py/applications/app/controllers/admin.py", line 447, in measurements_table
    'content': _measurement_actions, 'selected': False}])
  File "/vagrant/grand3/web2py/gluon/sqlhtml.py", line 3330, in __init__
    tablemap = dict(((f.tablename, f.table) for f in fieldmap.values()))
  File "/vagrant/grand3/web2py/gluon/sqlhtml.py", line 3330, in <genexpr>
    tablemap = dict(((f.tablename, f.table) for f in fieldmap.values()))
AttributeError: 'Expression' object has no attribute 'tablename'
```

**Comment by dburba** [comment]
My issue with TOTAL_ROWS is not resolved in Version 2.16.1-stable+timestamp.2017.11.14.05.54.25

```
Traceback (most recent call last):
  File "/vagrant/grand3/web2py/gluon/restricted.py", line 219, in restricted
    exec(ccode, environment)
  File "/vagrant/grand3/web2py/applications/app/controllers/admin.py", line 695, in <module>
  File "/vagrant/grand3/web2py/gluon/globals.py", line 419, in <lambda>
    self._caller = lambda f: f()
  File "/vagrant/grand3/web2py/gluon/tools.py", line 3982, in f
    return action(*a, **b)
  File "/vagrant/grand3/web2py/applications/app/controllers/admin.py", line 447, in measurements_table
    'content': _measurement_actions, 'selected': False}])
  File "/vagrant/grand3/web2py/gluon/sqlhtml.py", line 3433, in __init__
    tablemap = dict(((f.tablename, f.table) if isinstance(f, Field) else (f._table._tablename, f._table) for f in fieldmap.values()))
  File "/vagrant/grand3/web2py/gluon/sqlhtml.py", line 3433, in <genexpr>
    tablemap = dict(((f.tablename, f.table) if isinstance(f, Field) else (f._table._tablename, f._table) for f in fieldmap.values()))
AttributeError: 'NoneType' object has no attribute '_tablename'
```
